text/0032-dust-protection/0032-dust-protection.md

```diff
@@ -0,0 +1,114 @@
+                                                    <tr>
+                                                        <td>Address</td>
+                                                        <td>Array&lt;byte&gt;[32]</td>
+                                                        <td>The raw bytes of the Ed25519 address which is a BLAKE2b-256 hash of the Ed25519 public key.</td>
+                                                    </tr>
+                                                </table>
+                                            </details>
+                                        </td>
+                                    </tr>
+                                    <tr>
+                                        <td>Amount</td>
+                                        <td>uint64</td>
+                                        <td>The amount of tokens to deposit with this <i>SigLockedSingleOutput</i> output.</td>
+                                    </tr>
+                                </table>
+
+
+### The new protocol dust rule
+
+Let `A` be the address that should hold the dust outputs' balances. Let `M` be the sum of all SigLockedDustAllowanceOutputs' values on `A`.So the total allowed dust outputs on `A` is (`M`/1 MIOTA) * 100 rounded down to the nearest integer. So 100 outputs for 1 MIOTA deposited.
+
+The SigLockedDustAllowanceOutput must be greater or equal to 1MI.
+
+If while processing a milestone a new dust UTXO is created but there is not a sufficient amount already locked in SigLockedDustAllowanceOutputs then the encompassing message should be marked as `ignored`.
+
+If while processing a milestone an unspent SigLockedDustAllowanceOutput is consumed and as a result we have a violation on the number of allowed dust outputs then the encompassing message should be marked as `ignored`.
+
+The checks should happen after the entire message was processed. Messages that were previously applied by the current milestone are taken into account.
+
+
+# Drawbacks
+
+1. We can't have an address that will ever hold less than 1 MI. It will be a bit hard on testnet faucets but still workable.
+2. A service receiving micropayments may fail receiving them if it didn't consolidate dust outputs or raised the deposit for the address.
+
+# Rationale and alternatives
+
+Other solutions that were discussed:
+
+*Burning Dust* - allow dust to live for a limited amount of time on the ledger before removing it from total supply.
+
+*Sweeping dust into merkle tree* - Instead of storing all of the dust in the UTXO set, some nodes can accumulate the dust outputs to a merkle tree root. When the dust is spent, those nodes will require a merkle proof of inclusion.
+
+The first solution has some attack vectors that may burn dust that a user may have wanted to consolidate. Also solutions changing the supply can be quite controversial.
+
+The second solution, besides being more complicated, will relief the problem from one class of nodes but not all nodes.
```

Comment:
```suggestion
The second option is much more complicated as it introduces a completely new unlock mechanisms and requires the nodes to store the Merkle tree roots indefinitely.
```

Reply:
I think user-side added complexity should also be mentioned: a user need a way to be notified about the compression of his dust outputs into a certain merkle root, and need a way to efficiently query some permanode to obtain the corresponding audit path, bla bla. No need to be super-specific but just underline the fact that it exposes tons of complexities.